staging/src/k8s.io/apiserver/pkg/endpoints/discovery/root/discovery_test.go

```diff
@@ -0,0 +1,133 @@
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+var scheme = runtime.NewScheme()
+var codecs = runtimeserializer.NewCodecFactory(scheme)
+var serializer runtime.NegotiatedSerializer
+
+const discoveryPath = "/apis"
+const jsonAccept = "application/json"
+const aggregatedAccept = ";g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList"
+
+func init() {
+	// Add all builtin types to scheme
+	k8sscheme.AddToScheme(scheme)
+	apidiscoveryv2beta1.AddToScheme(scheme)
+	info, ok := runtime.SerializerInfoForMediaType(codecs.SupportedMediaTypes(), runtime.ContentTypeJSON)
+	if !ok {
+		panic("failed to create serializer info")
+	}
+	serializer = runtime.NewSimpleNegotiatedSerializer(info)
+}
+
+func fetchPath(handler http.Handler, path, accept string) string {
+	w := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", discoveryPath, nil)
+
+	// Ask for JSON response
+	req.Header.Set("Accept", accept)
+
+	handler.ServeHTTP(w, req)
+	return string(w.Body.Bytes())
+}
+
+type fakeHTTPHandler struct {
+	data string
+}
+
+func (f fakeHTTPHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	io.WriteString(resp, f.data)
+}
+
+func TestAggregationEnabled(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AggregatedDiscoveryEndpoint, true)()
+
+	unaggregated := fakeHTTPHandler{data: "unaggregated"}
+	aggregated := fakeHTTPHandler{data: "aggregated"}
+	rootHandler := NewRootAPIsHandler(serializer, unaggregated, aggregated)
+
+	testCases := []struct {
+		accept   string
+		expected string
+	}{
+		{
+			accept:   jsonAccept + aggregatedAccept,
+			expected: "aggregated",
+		}, {
+			accept:   jsonAccept,
+			expected: "unaggregated",
+		}, {
+			// Server should return the first accepted type
+			accept:   jsonAccept + aggregatedAccept + "," + jsonAccept,
+			expected: "aggregated",
+		},
+	}
+
+	for _, tc := range testCases {
+		body := fetchPath(rootHandler, discoveryPath, tc.accept)
+		assert.Equal(t, tc.expected, body)
+	}
+}
+
+func TestAggregationDisabled(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AggregatedDiscoveryEndpoint, false)()
+
+	unaggregated := fakeHTTPHandler{data: "unaggregated"}
+	aggregated := fakeHTTPHandler{data: "aggregated"}
+	rootHandler := NewRootAPIsHandler(serializer, unaggregated, aggregated)
+
+	testCases := []struct {
+		accept   string
+		expected string
+	}{
+		{
+			accept:   jsonAccept + aggregatedAccept,
+			expected: "unaggregated",
+		}, {
+			accept:   jsonAccept,
+			expected: "unaggregated",
+		}, {
+			// Server should return the first accepted type.
+			// If aggregation is disabled, the unaggregated type should be returned.
+			accept:   jsonAccept + aggregatedAccept + "," + jsonAccept,
+			expected: "unaggregated",
+		},
```

Comment:
Good, now add the same things for protobuf.

Reply:
We're using `responsewriters.WriteObjectNegotiated` in so protobuf support should come for free. I added the desired headers, this unit test is only supposed to check whether the proxying is to the correct location, so correctness check will not be here.

Reply:
I think an additional single unit test showing both json and protobuf are supported even a little but is still warranted. 

unit test would catch if someone were ever to remove `responsewriters.WriteObjectNegotiated` with a replacement that did not support all our intended content types

Reply:
I wasn't trying to say we shouldn't have unit tests for protobuf, rather this is probably not the best package to place it in.  `discovery -> aggregated -> handler_test.go` just feels like a better place to put it